pkg/util/vmopv1/vm.go

```diff
@@ -213,3 +216,123 @@ func IsClasslessVM(vm vmopv1.VirtualMachine) bool {
 func IsImagelessVM(vm vmopv1.VirtualMachine) bool {
 	return vm.Spec.Image == nil && vm.Spec.ImageName == ""
 }
+
+func SyncStorageUsageForNamespace(
+	ctx context.Context,
+	k8sClient client.Client,
+	namespace, storageClass string) (bool, <-chan error) {
+
+	var (
+		chanErr = make(chan error, 1)
+		objKey  = client.ObjectKey{
+			Namespace: namespace,
+			Name:      spqutil.StoragePolicyUsageName(storageClass),
+		}
+	)
+
+	syncFn := func() {
+		defer close(chanErr)
+
+		ctx := context.Background()
+
+		// Make sure the StoragePolicyUsage document exists.
+		var obj spqv1.StoragePolicyUsage
+		if err := k8sClient.Get(ctx, objKey, &obj); err != nil {
+
+			// Even if the SPU is not found, log the error to indicate that
+			// the function was enqueued but could not proceed.
+			chanErr <- fmt.Errorf(
+				"failed to get StoragePolicyUsage %s: %w", objKey, err)
+			return
+		}
+
+		var list vmopv1.VirtualMachineList
+		if err := k8sClient.List(
+			ctx,
+			&list,
+			client.InNamespace(namespace)); err != nil {
+
+			chanErr <- fmt.Errorf(
+				"failed to list VMs in namespace %s: %w", namespace, err)
+			return
+		}
+
+		var (
+			totalUsed     resource.Quantity
+			totalReserved resource.Quantity
+		)
+
+		for i := range list.Items {
+			vm := list.Items[i]
+
+			if !vm.DeletionTimestamp.IsZero() || vm.Status.Storage == nil {
+				// Ignore VMs being deleted or that are not reporting storage
+				// status.
+				continue
+			}
+
+			// Account for storage used by this VM.
+			if vm.Status.Storage.Unshared != nil {
+				totalUsed.Add(*vm.Status.Storage.Unshared)
+			}
+
+			// Account for storage marked as reserved by this VM.
```

Comment:
@SandeepPissay / @xing-yang, question for y'all,

To calculate the total space that could be used by a VM (i.e. reserved), we should be summing Committed and Uncommitted from https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.Summary.StorageSummary.html, right?

Reply:
The doc you pointed to says:

committed - Total storage space, in bytes, committed to this virtual machine across all datastores.
Essentially an aggregate of the property [committed](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.StorageInfo.UsageOnDatastore.html#committed) across all datastores that this virtual machine is located on.
uncommitted - Additional storage space, in bytes, potentially used by this virtual machine on all datastores.
Essentially an aggregate of the property [uncommitted](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.StorageInfo.UsageOnDatastore.html#uncommitted) across all datastores that this virtual machine is located on.


I think those properties will include any attached PVs as well. Also see my comment above on how we are using the provisioned capacity for storage quota accounting.

Reply:
If we use "committed" and "uncommitted", could we be double counting?  I see that there is another value named "unshared". Should we use that instead?

unshared - Total storage space, in bytes, occupied by the virtual machine across all datastores, that is not shared with any other virtual machine.

Reply:
> I think those properties will include any attached PVs as well. Also see my comment above on how we are using the provisioned capacity for storage quota accounting.

@SandeepPissay , yes, but as I noted elsewhere, we are subtracting these values.

Reply:
> If we use "committed" and "uncommitted", could we be double counting? I see that there is another value named "unshared". Should we use that instead?

@xing-yang ,

I don't think so. Unshared is for used. Committed + uncommitted is potential (i.e. reserved), minus the PVCs.

Reply:
Okay.